gateway/src/main/java/io/camunda/zeebe/gateway/ResponseMapper.java

```diff
@@ -276,22 +278,30 @@ public static SetVariablesResponse toSetVariablesResponse(
     return SetVariablesResponse.newBuilder().setKey(key).build();
   }
 
-  public static ActivateJobsResponse toActivateJobsResponse(
-      final long key, final JobBatchRecord brokerResponse) {
+  public static JobActivationResult toActivateJobsResponse(
```

Comment:
🔧 I'd like to see some Javadoc on this method, to clarify why the result contains a response **and** jobs to defer, and why a maxMessageSize is needed.

🔧 When writing that, we may choose to rename this maxMessageSize to maxResponseSize, as this better clarifies the parameter in the context of the method.

💭 I also wonder whether jobsToDefer is a fitting name in this method. In the context of the method, something like jobsExceedingSize, jobsThatDontFitInTheResponse, or something along those lines, may fit better semantically.

Reply:
Good points, updated!